x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/values_range_input.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+import { i18n } from '@kbn/i18n';
+import { EuiRange } from '@elastic/eui';
+
+type PropType<C> = C extends React.ComponentType<infer P> ? P : unknown;
+// Add ticks to EuiRange component props
+const FixedEuiRange = (EuiRange as unknown) as React.ComponentType<
+  PropType<typeof EuiRange> & {
+    ticks?: Array<{
+      label: string;
+      value: number;
+    }>;
+  }
+>;
+
+export const ValuesRangeInput = ({
+  value,
+  onChange,
+}: {
+  value: number;
+  onChange: (value: number) => void;
+}) => {
+  const MIN_NUMBER_OF_VALUES = 1;
+  const MAX_NUMBER_OF_VALUES = 100;
+
+  const [inputValue, setInputValue] = useState(String(value));
+  useDebounce(
+    () => {
+      const inputNumber = Number(inputValue);
+      if (inputNumber >= MIN_NUMBER_OF_VALUES && inputNumber <= MAX_NUMBER_OF_VALUES) {
```

Comment:
It works here because `Number('')` fallbacks to `0`, and `MIN_NUMBER_OF_VALUES` is > 0. But in the future it may change.
Just for clarity I'd add a new condition for the empty string (in case in the future somebody makes `0` a valid range end) and then will check for the range validity.